packages/access-client/readme.md

```diff
@@ -28,6 +51,78 @@ if (!(await store.exists())) {
+
+To set the current space, use [`agent.setCurrentSpace`](https://web3-storage.github.io/w3protocol/classes/_web3_storage_access.Agent.html#setCurrentSpace). Note that this must be done explicitly; creating an Agent's first Space does not automatically set it as the current space.
+
+### Space registration
+
+A newly-created Space must be registered with the w3up access service before it can be used as a storage location.
+
+To register a space, use [`agent.registerSpace`](https://web3-storage.github.io/w3protocol/classes/_web3_storage_access.Agent.html#registerSpace), which takes an email address parameter and registers the [current space](#managing-the-current-space) with the access service.
+
+Calling `registerSpace` will cause the access service to send a confirmation email to the provided email address. When the activation link in the email is clicked, the service will send the Agent a delegation via a WebSocket connection that grants access to the services included in w3up's free tier. The `registerSpace` method returns a `Promise` that resolves once the registration process is complete. Make sure to wrap calls to `registerSpace` in a `try/catch` block, as registration will fail if the user does not confirm the email (or if network issues arise, etc.).
+
+### Delegating to another Agent
+
+The Agent's [`delegate` method](https://web3-storage.github.io/w3protocol/classes/_web3_storage_access.Agent.html#delegate) allows you to delegate capabilities to another Agent.
+
+```js
+const delegation = await agent.delegate({
+  audience: 'did:key:kAgentToDelegateTo',
+  abilities: [
+    {
+      can: 'space/info',
+      with: agent.currentSpace()
+    }
+  ]
+})
+```
+
+Note that the receiving agent will need to [import the delegation](#importing-delegations-from-another-agent) before they will be able to invoke the delegated capabilities.
+
+### Importing delegations from another Agent
+
+The [`addProof` method](https://web3-storage.github.io/w3protocol/classes/_web3_storage_access.Agent.html#addProof) takes in a ucanto `Delegation` and adds it to the Agent's state Store. The proof of delegation can be retrieved using the Agent's [`proofs` method](https://web3-storage.github.io/w3protocol/classes/_web3_storage_access.Agent.html#proofs).
+
+
+The [`importSpaceFromDelegation` method](https://web3-storage.github.io/w3protocol/classes/_web3_storage_access.Agent.html#importSpaceFromDelegation) also accepts a ucanto `Delegation`, but it is tailored for "full delegation" of all Space-related capabilities. The delegated ability must be `space/*`, which is the "top" of the `space/` ability set. Use `importSpaceFromDelegation` in preference to `addProofs` when importing a full `space/*` delegation, as it also adds metadata about the imported Space to the Agent's `spaces` Map and persistent Store.
```

Comment:
The delegated ability must be "*" (top). I'm not sure why this is restricted right now @hugomrdias?

I'd avoid talking about the internal data structure (`Map`) here also, but obviously mention that it means the agent tracks it as a "space".

Reply:
The concepts here are hard to distinguish, meaning delegation vs space. What is a space ? It's a DID plus a delegation to do something with it. But you can have a delegation to just do 'store/add' in a space, does that mean you have a that space? No, you have the just that capability. So for now until we have a better way to design this.

Space === `{ can: '*', with: did:space, audience: did:agent }` and Proof === `{ can: 'less than *', with: did:space, audience: did:agent }`

And delegations === `{ can: 'anything agent can delegate', with: did:space, audience: did:someOtherAgent }`

So theres these 3 concepts here, 

1. **_Space_** meaning dids the agent has full capabilities, 
2. **_Proofs_** meaning all the delegations where the audience is the Agent 
3. **_Delegations_** meaning all the delegations the agent create to another agent.